### PR TITLE
Interview list filter by date

### PR DESCRIPTION
**Pull request by davidmurray**
for some reason, the SQL query `extract(epoch from created_at) = '2023-04-14'` did not work (no results). Had to switch to this solution: https://stackoverflow.com/questions/23335970/postgresql-query-between-date-ranges#comment112830851_48117316

UI:
image

### CODE REVIEW

**Comment by davidmurray** [comment]
Improved UI (commit [abe7cb6](https://github.com/chairemobilite/evolution/pull/290/commits/abe7cb69a6862101c7fcc030a5768424115e1546))
image

**Comment by davidmurray** [comment]
weird error in lint process:
"   8:20  error    "moment" is extraneous                     node/no-extraneous-import"

... but it is being used... Hm

**Comment by tahini** [comment]
Le pakcage 'moment' n'est probablement pas dans le package.json de evolution-frontend. Il faut le rajouter. Attention de bien mettre la même version qu'ailleurs!

**Comment by tahini** [comment]
le `extract epoch` retourne le timestamp a`la seconde (ou ms) près et n'est pas fait pour comparer avec du = à moins d,être extrêmement chanceux, mais si on compare avec des < ou >, alors ça devrait marcher. D'ailleurs un date range est probablement plus approprié ici. Le changement à la requête a peut-être aussi un impact sur d'autres calls à cette fonction qui utilisaient le timestamp?

File: packages/evolution-frontend/src/components/pageParts/validations/InterviewByDateFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    column: { filterValue, setFilter }
+}: FilterProps<InterviewListAttributes<CustomSurvey, CustomHousehold, CustomHome, CustomPerson>> & WithTranslation) => {
+    const [selectedDate, setSelectedDate] = React.useState<Date | null>(
+        !_isBlank(filterValue) && filterValue.value ? new Date(filterValue.value) : null
+    );
+
+    return (
+        <div style={{ display: 'flex', alignItems: 'center' }}>
+            <label htmlFor={'interviewDateSearchInput'}>{t('admin:interviewByDateFilter:title')}</label>
+            <DatePicker
```
Comment (tahini) [inline packages/evolution-frontend/src/components/pageParts/validations/InterviewByDateFilter.tsx:35]: See the RefreshInterviewerCache for a datePicker for a range

**Comment by davidmurray** [comment]
@tahini  fixed :) ready for review.
Now it accepts a date range

**Review by tahini — CHANGES_REQUESTED** [review]
Sequential tests now fail since the query has changed its format. Also, a db test should be added to cover this case too.

To run sequential tests, you need to run `yarn setup-test && yarn migrate-test && yarn test:sequential`

**Comment by tahini** [comment]
Also, if you change the format of the created_at date, you could also change the format of the updated_at, which is similar.

File: packages/evolution-frontend/package.json [hunk 2]
```diff
@@ -42,7 +42,7 @@
     "lodash": "^4.17.21",
     "node-fetch": "^2.6.7",
     "react": "^16.13.1",
-    "react-datepicker": "^3.1.3",
+    "react-datepicker": "^4.18.0",
```
[the remark this answers is not in the document]
Comment (davidmurray) [reply packages/evolution-frontend/package.json:45]: yes. Old version had a bug where the input field would only show the start date of the interval, whereas this version correctly shows the start and end dates

**Comment by davidmurray** [comment]
@tahini I re-worked the code to use unix timestamps now instead of just a date. This allows more specific filtering (unused now, but could be in the future).
This fixes the tests. I also added a new test to check if we can filter for a range of dates.

**Comment by davidmurray** [comment]
@tahini rebased to remove upgrade of react-datepicker :)

**Review by tahini — APPROVED** [review]
Thanks!

**Pull request merged**